game-core/src/main/java/games/strategy/engine/framework/GameDataManager.java

```diff
@@ -59,16 +61,35 @@ public static GameData loadGame(final InputStream is) throws IOException {
 
     final ObjectInputStream input = new ObjectInputStream(new GZIPInputStream(is));
     try {
-      final Version readVersion = (Version) input.readObject();
-      if (!ClientContext.engineVersion().isCompatibleWithEngineVersion(readVersion)) {
-        final String error =
-            "Incompatible engine versions. We are: "
-                + ClientContext.engineVersion()
-                + " . Trying to load game created with: "
-                + readVersion
-                + "\nTo download the latest version of TripleA, Please visit "
-                + UrlConstants.DOWNLOAD_WEBSITE;
-        throw new IOException(error);
+      final Object version = input.readObject();
+
+      if (version instanceof games.strategy.util.Version) {
+        throw new IOException(
+            String.format(
+                "Incompatible engine versions. We are: %s<br/>"
+                    + "Trying to load incompatible save game version: %s<br/>"
+                    + "To download an older version of TripleA,<br/>"
+                    + "please visit: <a href=%s>%s</a>",
+                ClientContext.engineVersion(),
+                ((games.strategy.util.Version) version).getExactVersion(),
+                UrlConstants.OLD_DOWNLOADS_WEBSITE,
+                UrlConstants.OLD_DOWNLOADS_WEBSITE));
+
+      } else if (!(version instanceof Version)) {
+        throw new IOException(
+            "Incompatible engine version with save game, "
+                + "unable to determine version of the save game");
```

Comment:
In short, the more detail we can get, the better.
Considering the reported issue: https://github.com/triplea-game/triplea/issues/5957, @ron-murhammer said: "Yeah we need a good error notification here as this will happen fairly often."

Without a good error message, I'm pretty positive a generic error message that does not really tell a user why the save game cannot be loaded would get reported as a bug. We do not have really any other way to tell a user to accurately that they need a 1.9 version. IMO, having the legacy class is a cost of the serialization technology used for save games. We introduced a feature regression when we removed the 'old jar' capability, keeping a legacy class around for a release is not a material cost to us.

Generally agree that we probably should encode the version as a string to keep the compatibility. Though, that is not what we have or had, it might be better to move directly to the project where we fix save game serialization and skip the intermediary step.